enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/service/CommonOtpService.java

```diff
@@ -107,8 +107,10 @@ protected OtpVerifyResponse verifyOtpCode(String processId, OwnerId ownerId, Str
         int maxFailedAttempts = commonOnboardingConfig.getOtpMaxFailedAttempts();
         if (OtpStatus.ACTIVE != otp.getStatus()) {
             logger.warn("Unexpected not active {}, process ID: {}", otp, processId);
+            auditService.audit(otp, "Unexpected not active {}, process ID: {}", otp, processId);
         } else if (failedAttempts >= maxFailedAttempts) {
             logger.warn("Unexpected OTP code verification when already exhausted max failed attempts, process ID: {}", processId);
+            auditService.audit(otp, "Unexpected OTP code verification when already exhausted max failed attempts, process ID: {}", processId);
```

Comment:
Maybe the issue was not clear enough. Let's discuss it with @romanstrobl. I would expect mainly auditing at https://github.com/wultra/enrollment-server/pull/781/files#diff-efdb8a9a5ee7fc0529c37e73c2ea88c9e8adedeefbcd321361d2becfecefa908R134

Reply:
Yes, this is exactly missing, auditing in case the verification fails and maximum attemps and timeout conditions have not been reached yet.